### PR TITLE
[HOTFIX#193] Fixed acceptProposal RPC for not properly cleaning exchanged cards from custom decks

### LINKED ISSUE #193: A seguito di uno scambio, le carte scambiate rimangono nel mazzo delle carte personalizzate dei vecchi proprietari, se già presenti in precedenza.
Come da titolo.

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR fixes the bug explained in issue #193. Also Refactored acceptProposal RPC for code cleanliness and moved decks updating to separate function `updateUserDecks`.

### CODE REVIEW

File: src/main/java/com/aadm/cardexchange/server/services/DeckServiceImpl.java [hunk 1]
```diff
@@ -322,33 +323,51 @@ public List<PhysicalCardWithEmailDealing> getListPhysicalCardWithEmailDealing(St
...
+                        throw new RuntimeException("DB ROLLBACK!");
+                    }
+                }
+                for (PhysicalCard pCard : exchanged_away_cards) {
+                    if (!deck.removePhysicalCard(pCard)) {
+                        throw new RuntimeException("DB ROLLBACK!");
+                    }
+                }
+            } else if (deckName.equals(WISHED_DECK)) {
+                Map<Integer, List<PhysicalCard>> lookUp = deck.getPhysicalCards().stream().collect(
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/server/services/DeckServiceImpl.java:353]: Nice to have the lookup table included in this method seeing that is used only here

File: src/main/java/com/aadm/cardexchange/server/services/ExchangeServiceImpl.java [hunk 2]
```diff
@@ -115,34 +115,8 @@ public boolean acceptProposal(String token, int proposalId) throws AuthException
...
-                        );
-                    if (receiverWishedDeck != null)
-                        receiverDecks.put(WISHED_DECK, DeckServiceImpl.removePhysicalCardsFromWishedDecksAfterExchange(
-                                receiverWishedDeck, proposal.getSenderPhysicalCards())
-                        );
-
-                    // save
-                    deckMap.put(proposal.getReceiverUserEmail(), receiverDecks);
-                    deckMap.put(proposal.getSenderUserEmail(), senderDecks);
+                    deckMap.put(proposal.getReceiverUserEmail(), DeckServiceImpl.updateUserDecks(deckMap.get(proposal.getReceiverUserEmail()), proposal.getSenderPhysicalCards(), proposal.getReceiverPhysicalCards()));
```
Comment (Matte0Sacc0) [inline src/main/java/com/aadm/cardexchange/server/services/ExchangeServiceImpl.java:118]: Awsome compaction

**Review by Matte0Sacc0 — APPROVED** [review]
A really great job

**Review by dstops — APPROVED** [review]
Great job, cod is clear and clean. The methods are now much more concise but well-structured.

**Pull request merged**